### DISCUSSION TITLE
quarkus-oidc management of properties with discovery

### DISCUSSION THREAD

**Discussion by egavard-lmfr** [quarkusio/quarkus#22806] [Q&A] [answered]
Hello,

I was wondering if, when discovery of the URLs are enabled thanks to a well-known URI and the proper properties, URLs set in properties such as JWKS_uri should act as overriding instead of being simples fallbacks.

I'm trying to manage the fact that our access_token are signed by a certificate that differs from our id_token.
This means we have a "custom" jwks_uri in our well-known for that.

I'd like to use the discovery, except for that specific parameter that I'd like to override. 
I believe that this should be the default behavior. 

Proposal:

When discovery is enabled, oidc properties are override instead of fallbacks.
Otherwise, they are fallbacks.

If so, I'd be pleased to make the developments for this feature :)

[a comment by an automated account is not shown]

**Comment by sberyozkin** [comment, marked as answer]
Hi @egavard-lmfr 

I think it makes sense and would not break anything - the users either use the discovery or disable it and manually set all the required properties, when both the discovered and manually configured properties are available then the preference should indeed be given to the ones set up locally.

Please go ahead with creating a PR, thanks

**Comment by sberyozkin** [comment]
@egavard-lmfr FYI, https://github.com/quarkusio/quarkus/pull/26762
